File: examples/principal-multiple/principal-multiple.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')


def statements():
    # doc-start
    s1 = statement.Sts() \
                .allow() \
                .to_assume_role() \
                .for_account('1234567890', '0987654321')

    # when you already have a list:
    accounts = ['1234567890', '0987654321']
    s2 = statement.Sts() \
                .allow() \
                .to_assume_role() \
                .for_account(*accounts)

    s3 = statement.Sts() \
                .allow() \
                .to_assume_role() \
                .for_user('1234567890', 'Bob', 'John')

    # when you already have a list:
    users = ['Bob', 'John']
    s4 = statement.Sts() \
                .allow() \
                .to_assume_role() \
                .for_user('1234567890', *users)
    # doc-end

    return [s1, s2, s3, s4]


all = statements()
out(all)
# deploy(all, type='assume') Disabled, because we need valid principals to test

File: examples/principal-multiple/principal-multiple.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import * as statement from '../../lib';

function getStatements() {
  function wrap() {
    // doc-start
    const s1 = new statement.Sts()
      .allow()
      .toAssumeRole()
      .forAccount('1234567890', '0987654321');

    // when you already have a list:
    const accounts = ['1234567890', '0987654321'];
    const s2 = new statement.Sts()
      .allow()
      .toAssumeRole()
      .forAccount(...accounts);

    const s3 = new statement.Sts()
      .allow()
      .toAssumeRole()
      .forUser('1234567890', 'Bob', 'John');

    // when you already have a list:
    const users = ['Bob', 'John'];
    const s4 = new statement.Sts()
      .allow()
      .toAssumeRole()
      .forUser('1234567890', ...users);
    // doc-end
    return [s1, s2, s3, s4];
  }
  return wrap();
}
const s = getStatements();
out(s);
// deploy(s); //Disabled, because we need valid principals to test
